### ISSUE TITLE
Documentation site

### ISSUE THREAD

**Issue by tmcw** [mapbox/mapbox.js#29]
The first shot at this is here: http://mapbox.com/mapbox.js/

@tristen do you have any thoughts on how to improve this? Basically the approach right now is (and this is prototyped, I haven't written the script to do it automatically)
1. Pull in node_modules of modules as usual
2. Get the `API.md` file out of each, copy to _posts and add YAML header
3. Serve a jekyll site with a straight list of them all
4. Build a TOC in JS

Ideally this will have the mapbox look, and may live in mapbox.github.com

**Comment by tristen** [comment]
Nice! I really like this approach and seems simple enough to script out. Saman is sketching out a new grid and typography set we should base the design of the documentation off of. When there is a good consensus there - I can start implementing this.

**Comment by tristen** [comment]
Ah - I really liked the sound of this approach but currently using the build script to generate docs makes its a little disabling to apply doc like styles to the site.

Like there are sections where it would be nice to style out more clearly a passage like this:

> This loads the information about a map on MapBox Hosting. The first argument can either be a full URL to a TileJSON file, like 
> http://a.tiles.mapbox.com/v3/tmcw.map-hehqnmda.jsonp, or a bare id, like tmcw.map-hehqnmda, which will get expanded to the > former.

In the passage above it would be nice to have a little heads up section on what TileJSON is and also a sort of "pro tip" note indicating you can just add in the id and mapbox.js will fill in the rest.

Also, not being able really add markup for syntax highlighting is a bummer. we could add `<pre><code data-language=''javascript">` or something but that doesn't work well for html which would just be rendered out.

> Build a TOC in JS

We can avoid having to do this if we don't generate the docs from API and STYLE. But API and STYLE should remain and we should follow the same formatting guidelines you have established here. We just unfortunately have a little duplication here.

**Comment by tristen** [comment]
The other point:

> Serve a jekyll site with a straight list of them all

Having worked on some of this on Friday - it felt a little too long going down this route and static pages are fast to generate anyways. Splitting them up into separate pages seems to feel better.

**Comment by tmcw** [comment]
Any alternative suggestions? I'm not sure that these are all really that valid. For instance, for syntax highlighting we can use prettify in js which I've used in Wax and elsewhere.

The jekyll site can be multiple pages, and we can have inter-doc links that just don't work as well in documentation pages.

Would this be more doable if we just abandoned the autogenerated approach and made this more manual? I'm trying to push out a dev release to the team like today and the rest is mostly ready to go, so whatever works works.
